src/stream/src/task/stream_manager.rs

```diff
@@ -707,16 +699,18 @@ impl LocalStreamManagerCore {
         handle.abort();
     }
 
-    /// `drop_all_actors` is invoked by meta node via RPC once the stop barrier arrives at all the
-    /// sink. All the actors in the actors should stop themselves before this method is invoked.
+    /// `drop_all_actors` is invoked by meta node via RPC.
     fn drop_all_actors(&mut self) {
         for (actor_id, handle) in self.handles.drain() {
-            self.context.retain_channel(|&(up_id, _)| up_id != actor_id);
-            self.actor_monitor_tasks.remove(&actor_id).unwrap().abort();
-            self.actors.remove(&actor_id);
-            // Task should have already stopped when this method is invoked.
+            tracing::debug!("force stopping actor {}", actor_id);
             handle.abort();
         }
+        self.actors.clear();
+        self.context.clear_channels();
+        if let Some(stack_trace_manager) = self.stack_trace_manager.as_mut() {
+            std::mem::take(stack_trace_manager);
```

Comment:
This fixes the occupation error in async stack trace after recovery. cc @Little-Wallace